tools/lib/makestuff.js

```diff
@@ -0,0 +1,34 @@
+const fs = require("fs");
+const CleanCSS = require('clean-css');
+const path = require('path');
+const _        = require('lodash');
+const config = require("../build_config");
+const del      = require('del');
+
+async function clean(directory) {
+  del.sync([directory])
+  fs.mkdirSync(directory, {recursive: true});
+};
+
+function install(file, dest=file) {
+  fs.copyFileSync(file, `${process.env.BUILD_DIR}/${dest}`);
+}
+
+function install_cleancss(file, dest) {
+  let content = fs.readFileSync(file, {encoding: "utf8"});
+  let out = new CleanCSS(config.clean_css).minify(content).styles;
+  fs.writeFileSync(`${process.env.BUILD_DIR}/${dest}`, out);
+}
+
+function mkdir(dirname) {
```

Comment:
I don't like the name very much, because it looks like a general-purpose function, but it's works for BUILD_DIR only. Something like `build_mkdir` would be better, although I'm not totally happy with this name either.

Reply:
Node already has a built in `mkdir` sync and unsync versions... I think these are build tool internals and if you want to know the specifics you read them.  i think contextual when you read the build scripts and see the build output it's pretty clean everything is working in terms of the build directory, so I'm not sure I see this as a real issue.

Reply:
So, willing to let this slide?  the one thing I can think of would be to namespace a little more:

```
const build = require("./lib/builddir_make");

...
build.mkdir()
```

Personally I think it's a bit verbose and doesn't really add that much clarity.

Reply:
Yeah, I guess I see the slight issue with the name overlap now... something to think about next time we rewrite the build system. :-)